docs/visualization.md

```diff
@@ -0,0 +1,84 @@
+        * Consider the following simplified scenario to illustrate this approach.
+
+        ![picture alt](Approach2.png "Approach 2")
+
+    
+    * **Approach 4:**
+        * Similar to Approach 1. The difference lies in the fact that we represent messages as smaller rectanges.
+        * The message rectangles move from sender agent to receiver agent.
+
+    * **Approach 5:**
+        * Similar to Approach 2. The difference lies in the fact that we represent messages as smaller rectanges.
+        * The message rectangles move from sender agent to receiver agent.
+    * **Approach 6:**
+        * Each stage is represented by columns. Similar to github project boards, the orders can be moved from one column to another. 
+        * The agents cannot be visualised here, neither can the messages be tracked. But it gives a overall progress of how much a particular orders has been processed. More useful for customers.
+
+* Approach-1 would suffice our requirement quite well and is readily available.
+
+
+### Visualization Frameworks :
+
+* Depending on the material on which we render our visualization we divided this section into two types.
+
+* ** Visualization options with X Window options :**
+
+    * If the machine on which testing is being done supports X Windows, we can use available Java frameworks for rendering the agents and the interactions taking place between them.
+
+    * Some widely used X Windows based frameworks like Swing together with Abstract Window Toolkit (AWT) can be used to generate simple animations on the X Windows.
+
+    * Sniffer tool from tilab ([tilab](http://jade.tilab.com/documentation/tutorials-guides/sniffer/introduction/))
+
+* ** Visualization options using console : **
+
+    * If the machine on which testing is being done does not support X Windows, (maybe like travis on which automated testing is being done), then there are not many options readily available on Java.
+
+    * There are frameworks for converting images to ascii art to be displayed on the console but that is not what we need.
+
+    * There are frameworks/codebases for ascii 2d console games. However tweaking them for our use case is a huge project in itself. ([roguebasin](http://roguebasin.roguelikedevelopment.org/index.php?title=Main_Page#Java_Roguelike_Development_Guide))
+
+    * One approach might be to implement a simple stdout print based approach for displaying the predefined shapes on the console.
+
+### Visualization Agent Implementation :
+
+* Irrespective of the material on which the visualization is presented, the approach used for coveying the information from the Bakery agents to the Visualization agent is illustrated below.
+    * All the Agents in the bakery will update their status to a file (file format can be defined later).
```

Comment:
How will you handle read/write problems of multiple agents at the same time?

Reply:
According to my understanding, the synchronization between read/write from different agents is outside the scope of the visualization agent. Protection mechanisms like mutex/semaphores can be used in the individual agents whenever they want to write to the shared file. We are yet to discuss this aspect in detail. I will make a note of it to discuss this in our next team meeting.

Reply:
Followup Issue : https://github.com/HBRS-MAAS/ws18-project-right-brothers/issues/31